elastic/fleet-server
# chore(3379): Workspace and lsp config

- Added go.work and go.work.sum files
- Added information about lsp configuration in README.md



## What is the problem this PR solves?

I was running into lsp errors while trying to work on e2e tests. The details of the errors are outlined in the PR itself. This made working on the testing package quite difficult as I was not getting any sort of completions or suggestions.  

## How does this PR solve the problem?

One of the errors was due to the nested modules. Configuring go workspaces resolves this issue. This config can be found in `go.work`. The second issue was due to the build constraints (eg. `//go:build e2e`). The solution for this is explained in the updated README. 

## How to test this PR locally

Please make sure that `go.work` configuration works for you. With the changes in this PR, confirm that you are still getting code suggestions/completions in all the modules in the repo.

## Design Checklist




## Checklist





## Related issues
- Closes #3379
